musikcube.spec

```diff
@@ -0,0 +1,40 @@
+%define name musikcube
+%define build_timestamp %{lua: print(os.date("%Y%m%d"))}
+%define version 0.31.0
+Name: %{name}           
+Version: %{version}     
+Release: %{build_timestamp}%{dist}
+Summary: A cross-platform, terminal-based audio engine, library, player and server written in C++
+Source0: https://github.com/clangen/musikcube/archive/%{version}.tar.gz#/%{name}-%{version}.tar.gz
+License: BSD-3-Clause
+Packager: David Muckle <dvdmuckle@dvdmuckle.xyz>
+BuildRequires: gcc-c++ cmake boost-devel libogg-devel libvorbis-devel flac-devel faad2-devel ncurses-devel zlib-devel alsa-lib-devel pulseaudio-libs-devel libcurl-devel libmicrohttpd-devel lame-devel make
+Requires: boost libogg libvorbis flac ncurses zlib alsa-lib pulseaudio-libs libcurl libmicrohttpd lame
+Recommends: faad2
+
+%description
+
+A cross-platform, terminal-based audio engine, library, player and server written in C++
+%global debug_package %{nil}
+%prep
+%autosetup -n %{name}-%{version}
+
+
+%build
+#cmake -DCMAKE_INSTALL_PREFIX:PATH=%{buildroot}%{_prefix} .
+cmake .
+make -j2
+
+
+%install
+mkdir %{buildroot}%{_prefix} -p
+make install DESTDIR=%{buildroot}
+
+%clean
+rm -rf %{buildroot}
+
+%files
+%{_prefix}/local/bin/musikcube
```

Comment:
I don't know anything about RPM packaging; does this section specify where the files are to be installed? If so, is it correct/idiomatic to have `/local/` after `%{_prefix}`? Seems like the `/local/` section should be part of the path prefix variable.

Reply:
This was a bug I ran into, for some reason musikcube _really_ wanted to install to `/usr/local` instead of just `/usr`. This was only with RPM packages. The %files section just keeps track of where files are, in the case someone wants to know what package supplies what files, or remove the package and thus the files listed. You're right about the variable, there really should be one for `/usr/local`, however I wasn't able to find one.

Reply:
Hmm, interesting. What happened when you tried to install musikcube outside of `/usr/local`? I'm wondering if the `musikcube.in` file wasn't properly processed, and the path wasn't substituted correctly.

The installation should look as follows:

`<prefix>/share/musikcube/musikcube` <- actual binary
`<prefix>/share/musikcube/plugins`
`<prefix>/share/musikcube/themes`
`<prefix>/share/musikcube/locales`

and then:

`<prefix>/bin/musikcube` <- a small shell script wrapper generated using `musikcube.in`

Reply:
For some reason, `cmake .` and then `make && make install` places everything into `/usr/local`, only when building an rpm. I'll test further because this is very strange behavior.

Reply:
CMake may use some sort of OS default unless the prefix is explicitly specified. Have you tried the following?

`cmake . -DCMAKE_INSTALL_PREFIX=/usr ...`

Reply:
Yeah, `cmake .` sets `CMAKE_INSTALL_PREFIX` to `/usr/local`. I just fixed it so it will set to install to `/usr`.

Reply:
👍